vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
@tbohn: No, you wrote:

`(*veg_hist)[i] = (veg_hist_struct *) calloc(nveg, sizeof(veg_hist_struct));`

which is how I would have written it as well. However, we want to get rid of all the type info on the right hand side, which ends up being a bit awkward. I think:

`(*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i])));`

is correct, but I'd have to think about it